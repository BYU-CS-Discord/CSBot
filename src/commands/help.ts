import { appVersion, repo } from '../constants/meta';
import { EmbedBuilder } from 'discord.js';

export const help: GlobalCommand = {
	name: 'help',
	description: 'Prints useful info about the bot',
	requiresGuild: false,
	async execute({ reply }) {
		const embed = new EmbedBuilder()
			.setTitle('Help')
			.setDescription(
				`To see the list of commands, type \`/\` in your message field.\n[Visit our GitHub](${repo.href}) for more details!`
			)
			.setFooter({ text: `v${appVersion}` });

		await reply({
			embeds: [embed],
			ephemeral: true,
		});
	},
};
