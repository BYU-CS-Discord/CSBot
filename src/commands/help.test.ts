import type { CommandContext } from '../CommandContext';
import { help } from './help';

describe('help', () => {
	test("it's just a Hello World", async () => {
		const mockReply = jest.fn();
		const context = {
			reply: mockReply,
		} as unknown as CommandContext;

		await expect(help.execute(context)).resolves.toBeUndefined();

		expect(mockReply).toHaveBeenCalledOnce();
		expect(mockReply).toHaveBeenCalledWith('Hello, world!');
	});
});
