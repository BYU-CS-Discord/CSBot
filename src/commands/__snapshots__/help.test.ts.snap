// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help presents an ephemeral embed with general info 1`] = `
{
  "description": "To see the list of commands, type \`/\` in your message field.
[Visit our GitHub](https://github.com/BYU-CS-Discord/CSBot) for more details!",
  "footer": {
    "icon_url": undefined,
    "text": "vX.X.X",
  },
  "title": "Help",
}
`;
